mindsdb/integrations/handlers/notion_handler/notion_handler.py

```diff
@@ -0,0 +1,219 @@
+    target: str = None
+    ft_api_info: dict = None
+    ft_result_stats: dict = None
+    runtime: str = None
+    notion_api_token: str = None
+
+    class Config:
+        # for all args that are not expected, raise an error
+        extra = Extra.forbid
+
+
+class NotionHandler(APIHandler):
+    name = "notion"
+
+    def __init__(self, name: str, **kwargs):
+        """constructor
+        Args:
+            name (str): the handler name
+        """
+        super().__init__(name)
+
+        args = kwargs.get("connection_data", {})
+        self.connection_args = {}
+
+        handler_config = Config().get("notion_handler", {})
+        token_name = ["notion_api_token", "api_token"]
+        for k in token_name:
+            if len(self.connection_args.keys()) > 0:
+                self.connection_args[k] = self.connection_args.get(
+                    list(self.connection_args.keys())[0]
+                )
+            if k in args:
+                self.connection_args[k] = args[k]
+            elif f"NOTION_{k.upper()}" in os.environ:
+                self.connection_args[k] = os.environ[f"NOTION_{k.upper()}"]
+            elif k in handler_config:
+                self.connection_args[k] = handler_config[k]
+
+        self.api = None
+        self.is_connected = False
+
+        notion_database_data = NotionDatabaseTable(self)
+        notion_pages_data = NotionPagesTable(self)
+        notion_comments_data = NotionCommentsTable(self)
+        notion_blocks_data = NotionBlocksTable(self)
+
+        self._register_table("database", notion_database_data)
+        self._register_table("pages", notion_pages_data)
+        self._register_table("blocks", notion_blocks_data)
+        self._register_table("comments", notion_comments_data)
+
+    def connect(self, args=None, **kwargs):
+        api_token = self.connection_args["api_token"]
+        notion = Client(auth=api_token)
+        return notion
+
+    def check_connection(self) -> StatusResponse:
+        response = StatusResponse(False)
+
+        try:
+            api = self.connect()
+
+            response.success = True
+
+        except Exception as e:
+            response.error_message = (
+                f"Error connecting to Notion api: {e}. Check notion_api_token"
+            )
+            log.logger.error(response.error_message)
+
+        if response.success is True and not self.connection_args["api_token"]:
+            try:
+                api = self.connect()
+
+            except Exception as e:
+                keys = "notion_api_token"
+                response.error_message = (
+                    f"Error connecting to Notion api: {e}. Check" + ", ".join(keys)
+                )
+                log.logger.error(response.error_message)
+
+                response.success = False
+
+        if response.success is False and self.is_connected is True:
+            self.is_connected = False
+
+        return response
+
+    def _apply_filters(self, data, filters):
```

Comment:
This method does not seem to do anything.

Reply:
If it is not in use, let's remove it.

Reply:
yes it is being used in the [call_notion_api](https://github.com/Mr-Destructive/mindsdb-dev/blob/a7b26068153fcd64daf79e4f70e2642f8f419f14/mindsdb/integrations/handlers/notion_handler/notion_handler.py#L196-L197).